### ISSUE TITLE
A couple broken features in 11.02

### ISSUE THREAD

**Issue by Themurloc** [exochron/ToyBoxEnhanced#61]
First off, and probably most problematic, is that the search function no longer works as of the latest patch. It now returns a LUA error without filtering the toybox to the search terms.

```
Message: Interface/AddOns/ToyBoxEnhanced/Filters.lua:24: attempt to call global 'GetSpellDescription' (a nil value)
Time: Fri Aug 16 12:45:25 2024
Count: 14
Stack: Interface/AddOns/ToyBoxEnhanced/Filters.lua:24: attempt to call global 'GetSpellDescription' (a nil value)
[string "@Interface/AddOns/ToyBoxEnhanced/Filters.lua"]:24: in function <Interface/AddOns/ToyBoxEnhanced/Filters.lua:12>
[string "@Interface/AddOns/ToyBoxEnhanced/Filters.lua"]:253: in function `FilterToys'
[string "@Interface/AddOns/ToyBoxEnhanced/ToyBoxEnhanced.lua"]:166: in function <Interface/AddOns/ToyBoxEnhanced/ToyBoxEnhanced.lua:138>

Locals: itemId = 147310
searchString = "searchtest"
_ = 147310
name = "Crate of Bobbers: Floating Totem"
_ = "Floating Totem"
spellId = 240802
(*temporary) = nil
(*temporary) = 240802
(*temporary) = "attempt to call global 'GetSpellDescription' (a nil value)"
ADDON = <table> {
 initialized = true
 settings = <table> {
 }
 UI = <table> {
 }
 Events = <table> {
 }
 TOYS_PER_PAGE = 18
 db = <table> {
 }
 L = <table> {
 }
 DataProvider = <table> {
 }
}
```

Also, the filters menu appears to be missing several strings that tell you what the filters are.
![image](https://github.com/user-attachments/assets/c553de6e-7b97-4e20-8de0-c187d81ebc8b)

Your other addon, mountjournalenhanced, appears to be A-OK and having none of the same issues. Just this one

**Comment by Fluke667** [comment]
ToyBoxEnhanced/SettingsPanel.lua:20: attempt to call global 'GetAddOnMetadata' (a nil value)
[string "@ToyBoxEnhanced/SettingsPanel.lua"]:20: in function <ToyBoxEnhanced/SettingsPanel.lua:14>
[string "@ToyBoxEnhanced/SettingsPanel.lua"]:46: in function <ToyBoxEnhanced/SettingsPanel.lua:45>
[string "=[C]"]: ?
[string "@Blizzard_SharedXMLBase/CallbackRegistry.lua"]:144: in function <...eBlizzard_SharedXMLBase/CallbackRegistry.lua:143>
[string "=[C]"]: ?
[string "@Blizzard_SharedXMLBase/CallbackRegistry.lua"]:147: in function `TriggerEvent'
[string "@ToyBoxEnhanced/ToyBoxEnhanced.lua"]:159: in function <ToyBoxEnhanced/ToyBoxEnhanced.lua:154>

Locals:
frame = Frame {
}
L = <table> {
 FILTER_SECRET = "Hidden in game"
 RANDOM_TOY_TITLE = "Use Random Favorite Toy"
 Toys = "Toys"
 COMPARTMENT_TOOLTIP = "|cffeda55fLeft-Click|r to toggle showing the Toy Box.
|cffeda55fRight-Click|r to open addon options."
 FILTER_HIDDEN_MANUAL = "Hidden by me"
 SORT_FAVORITES_FIRST = "Favorites First"
 SETTING_SEARCH_IN_DESCRIPTION = "Search also in toy description"
 FAVOR_PER_CHARACTER = "Per Character"
 RANDOM_TOY_LOCKED = "Please favor at least one collected toy to unlock."
 TASK_FAVOR_START = "[TBE] Reapplying stars all over your toys. Please wait a few seconds until I'm finished."
 FILTER_ONLY = "only"
 Usable = "Usable"
 SETTING_CURSOR_KEYS = "Enable Left&Right keys to flip pages"
 SORT_UNOWNED_AFTER = "Unowned at Last"
 SORT_REVERSE = "Reverse Sort"
 TASK_END = "[TBE] Phew! I'm done."
 FILTER_ONLY_TRADABLE = "Only tradable"
 FILTER_ONLY_LATEST = "Only latest additions"
 SETTING_FAVORITE_PER_CHAR = "Favorite toys per character"
 RANDOM_TOY_DESCRIPTION = "The toy will be chosen randomly from your favorites."
 FAVOR_DISPLAYED = "All Displayed"
 Reset filters = "Reset filters"
}
titleFont = FontString {
}
(*temporary) = FontString {
}
(*temporary) = nil
(*temporary) = "ToyBoxEnhanced"
(*temporary) = "Title"
(*temporary) = "attempt to call global 'GetAddOnMetadata' (a nil value)"
ADDON = <table> {
 settings = <table> {
 }
 UI = <table> {
 }
 Events = <table> {
 }
 TOYS_PER_PAGE = 18
 db = <table> {
 }
 L = <table> {
 }
 DataProvider = <table> {
 }
}
ADDON_NAME = "ToyBoxEnhanced"

**Comment by exochron** [comment]
fixed with version 2.26
